src/main/java/com/ecore/roles/service/impl/RolesServiceImpl.java

```diff
@@ -0,0 +1,61 @@
+package com.ecore.roles.service.impl;
+
+import com.ecore.roles.exception.ResourceExistsException;
+import com.ecore.roles.exception.ResourceNotFoundException;
+import com.ecore.roles.model.Role;
+import com.ecore.roles.repository.MembershipRepository;
+import com.ecore.roles.repository.RoleRepository;
+import com.ecore.roles.service.MembershipsService;
+import com.ecore.roles.service.RolesService;
+import lombok.NonNull;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.UUID;
+
+@Log4j2
+@Service
+public class RolesServiceImpl implements RolesService {
+
+    public static final String DEFAULT_ROLE = "Developer";
+
+    private final RoleRepository roleRepository;
+    private final MembershipRepository membershipRepository;
+    private final MembershipsService membershipsService;
+
+    @Autowired
+    public RolesServiceImpl(
+            RoleRepository roleRepository,
+            MembershipRepository membershipRepository,
+            MembershipsService membershipsService) {
+        this.roleRepository = roleRepository;
+        this.membershipRepository = membershipRepository;
+        this.membershipsService = membershipsService;
+    }
+
+    @Override
+    public Role CreateRole(@NonNull Role r) {
```

Comment:
Please don't use shortcuts like both "Role r" and "UUID rid" in the methods parameters. Use full names, if it's not too long like both "Role role" and "UUID id".